### PR TITLE
Migrate to typescript

### PR DESCRIPTION
**Pull request by Nexite**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,72 +1,123 @@
...
-    "@codeday/eslint-config": "^1.3.0",
-    "@codeday/topocons": "^1.5.0",
-    "@storybook/addon-a11y": "^6.1.20",
-    "@storybook/addon-actions": "^6.1.20",
-    "@storybook/addon-docs": "^6.1.20",
-    "@storybook/addon-essentials": "^6.1.20",
-    "@storybook/addon-links": "^6.1.20",
-    "@storybook/addons": "^6.1.20",
-    "@storybook/react": "^6.1.20",
-    "@svgr/cli": "^5.4.0",
```
Comment (tylermenezes) [inline package.json:52]: svgr was needed for generating Atom/Logo, is this handled some other way now?

File: package.json [hunk 2]
```diff
@@ -1,72 +1,123 @@
...
-    "babel-loader": "^8.1.0",
-    "babel-plugin-module-resolver": "^4.0.0",
-    "copyfiles": "^2.4.1",
-    "create-index": "^2.5.0",
```
Comment (tylermenezes) [inline package.json:56]: create-index was also used in Atom/Logo

File: package.json [hunk 3]
```diff
@@ -1,72 +1,123 @@
...
-  },
-  "peerDependencies": {
-    "@codeday/topocons": "^1.3.0",
-    "prop-types": "^15.x",
-    "react": ">= 16.8",
-    "react-dom": ">= 16.8"
-  },
-  "optionalDependencies": {
-    "react-pdf": "^4.2.0",
-    "react-syntax-highlighter": "^13.2.1"
```
Comment (tylermenezes) [inline package.json:70]: These was previously marked as optional to reduce the size of node_modules for people not using the Deck and Codeblock components, as they were quite large. Do they need to be required?

**Review by tylermenezes — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (Nexite) [reply package.json:70]: Made these optional again

**Pull request merged**